### ISSUE TITLE
Question Names Not Shown to Students

### ISSUE THREAD

**Issue by gchallen** [PrairieLearn/PrairieLearn#1149]
Not sure why this is. Question names are shown in the instructor view, but not to students. For a long exam, question names may help students navigate and allocate their time:

<img width="1116" alt="screen shot 2018-05-07 at 11 35 19 am" src="https://user-images.githubusercontent.com/445443/39713024-c9d805a0-51ea-11e8-94a1-64da820324f7.png">

**Comment by coatless** [comment]
@gchallen issue here is some students tell their friends about the question contents based on the name. I think that's why the feature was disabled.

**Comment by mwest1066** [comment]
@gchallen yep, this is intentional. Question names are explicitly disabled in student exam mode to make collaboration harder.

**Comment by coatless** [comment]
Let's say **Question 3** is a _really_ difficult question. If we have the name of **"Some Crazy Difficult Question"**, then students are likely to say, "You need to do _XYZ_ on **Some Crazy Difficult Question**". Thus, when they open up their exam, the first question they will attempt is the **Some Crazy Difficult** question as they have prepared for it. Consider viewing this in the light of writing formulas on the top of an exam sheet the second the exam is handed out.

In comparison, if we have the question named **Question 3**, which could really be **Question 6** on another student's exam, then the difficulty in revealing and planning for each item increases. Unfortunately, as you have pointed out, the cost of this system is sacrificing the ability to easily move around with the gain being the randomization being harder to defeat.

With this in mind, I should point out that in exam mode the question's name is _revealed_ on the question page. This might need to get changed...

![screen shot 2018-05-09 at 11 37 01 am](https://user-images.githubusercontent.com/833642/39827363-5aa03e8e-537d-11e8-93c8-45398d47bd44.png)

**Comment by gchallen** [comment]
@coatless: I still don't really think that this makes a meaningful difference, particularly in the common case for quizzes that don't have a huge number of questions. Modulo the frontend delays, browsing an exam of ~10 questions to find the one that you've prepared to answer doesn't take a sufficient amount of time to discourage this behavior.

I understand the tradeoff: eliding the question names prevents this kind of attack, but also prevents honest students from easily navigating the exam. I just don't think the small decrease in the attack surface is worth the loss of navigability. Others may feel differently, which is why it would be nice if ths was configurable.

**Comment by coatless** [comment]
Revisiting this around a year later...

Any objection to adding a JSON toggle key to `infoAssessment.json` to switch between

```
Question <%= instance_question.question_number %>
```

And

```
<%= instance_question.question_title %>
```

Note: `question_title` is still present.

https://github.com/PrairieLearn/PrairieLearn/blob/3cebe3d009a672360dedd5799a27712e560b06fd/pages/studentAssessmentInstanceExam/studentAssessmentInstanceExam.ejs#L179

https://github.com/PrairieLearn/PrairieLearn/blob/3cebe3d009a672360dedd5799a27712e560b06fd/pages/studentAssessmentInstanceExam/studentAssessmentInstanceExam.sql#L7

**Comment by coatless** [comment]
Should the set options for enabling this appear in the definition of `type`

https://github.com/PrairieLearn/PrairieLearn/blob/66ebe11509957c4ef2c569e23f7b857443e14b68/schemas/schemas/infoAssessment.json#L20-L26

Or should this be set as an option in `accessRule`? 

https://github.com/PrairieLearn/PrairieLearn/blob/66ebe11509957c4ef2c569e23f7b857443e14b68/schemas/schemas/infoAssessment.json#L88

Alternatively, should a third `type` be introduced? 

Thoughts? @mwest1066 @nwalters512 @trombonekenny

**Comment by daviddalpiaz** [comment]
@coatless For that `type` are you thinking a `Quiz` that has:

- question names displayed
- ability to not show correct answer for incorrect submission
- exam like grading (but maybe also the ability for unlimited attempts like homework)

Because that's exactly what I'm looking for.

**Comment by goldwasser** [comment]
Just coming to this issue as new user in 2024.  While I can understand that some instructors would prefer to have the generic names, clearly some instructors would prefer for their students to be able to see the topics of questions from the navigation page so they can be more informed test takers.   So seems like this should be made a configurable setting, leave the default to match the current behavior, but allow instructor to override that presumably with some flag within infoAssignment.json.  Natural place could be within the "questions" list, perhaps with "displayedName":"Bubble sort", with default being unnamed.

**Comment by nwalters512** [comment]
This should be considered a part of https://github.com/PrairieLearn/PrairieLearn/issues/3347.

@goldwasser it sounds like you want to customize the question name on a per-assessment basis. Would it be sufficient to display the question's title as defined in the question's JSON file?

**Comment by firasm** [comment]
Good discussion everyone, always good to read and get different perspectives.

I also agree that if this was a customizable option per assessment that would solve this for me, and the title in the JSON file would be great.

I also think that (as was pointed out earlier) that if the option was set to hide the title from the assessments page, it should also be hidden on the question page (stealing @coatless' screenshot) and replaced with "Question N".

![Image](https://github.com/user-attachments/assets/9bed92b5-647d-44a7-ad28-d4d13e64a0a9)

**Comment by jonatanschroeder** [comment]
I believe this was resolved in #14531.
